oci/layout/oci_transport.go

```diff
@@ -228,7 +247,75 @@ func (ref ociReference) NewImageDestination(ctx context.Context, sys *types.Syst
 
 // DeleteImage deletes the named image from the registry, if supported.
 func (ref ociReference) DeleteImage(ctx context.Context, sys *types.SystemContext) error {
-	return errors.New("Deleting images not implemented for oci: images")
+
+	// Get the manifest for the image
+	manifestDescriptor, err := ref.getManifestDescriptor()
+	if err != nil {
+		return err
+	}
+
+	manifest, err := ref.getManifest(manifestDescriptor)
+	if err != nil {
+		return err
+	}
+
+	// Get all the layers used by all other images
+	index, err := ref.getIndex()
+	if err != nil {
+		return err
+	}
+	layersUsedByOtherImages := make(map[digest.Digest]bool, len(index.Manifests))
+	for _, v := range index.Manifests {
+		if v.Digest != manifestDescriptor.Digest {
+			otherImageManifest, err := ref.getManifest(v)
+			if err != nil {
+				return err
+			}
+			for _, layer := range otherImageManifest.Layers {
+				layersUsedByOtherImages[layer.Digest] = true
+			}
+		}
+	}
+
+	// Delete all blobs
+	blobsToDelete := make([]digest.Digest, 0, len(manifest.Layers))
+	for _, layer := range manifest.Layers {
+		if !layersUsedByOtherImages[layer.Digest] {
+			blobsToDelete = append(blobsToDelete, layer.Digest)
+		}
+	}
+	for _, digest := range append(blobsToDelete, manifest.Config.Digest, manifestDescriptor.Digest) {
```

Comment:
I think the config an manifest entries also need to be checked against `…UsedByOtherImages`.

Reply:
I'll add that too; that's quite cheap and provides an extra level of safety.